### PR TITLE
Fix xmldb:store for xs:anyURI

### LINKED ISSUE #1456: xmldb:store fails on remote binary resources
### What is the problem

The following query worked in eXist 2.2 but fails on eXist 3.x:

```xquery
xquery version "3.0";

xmldb:store(
    '/db', 
    'image.jpg', 
    xs:anyURI('https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png'), 
    'image/png'
)
```

The error in eXide in eXist 3.3.0-DEVELOP:

> exerr:ERROR IOException while reading from URL: https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png [at line 3, column 1]

In my application (where I first saw this), my try-catch block's `$err:description` reports this before the IOException:

> java:java.nio.file.FileAlreadyExistsException module: (114,35) /Users/joe/workspace/exist/tools/jetty/tmp/existDBS396998373878243953.xml.

But no such file exists in that location.

It looks like the last change to xmldb:store was @adamretter's commit of Oct 15, 2016 that changed src/org/exist/xquery/functions/xmldb/XMLDBStore.java (starting around ~line 286): https://github.com/eXist-db/exist/commit/c0b66cecad66cd6f00c91d12301914f65e42cf39#diff-1ae2fca2d57a1afd1fa116bb2548908e.

### What did you expect

I expected xmldb:store to allow remote binary resources to be stored, as stated in the function documentation. (See http://exist-db.org/exist/apps/fundocs/view.html?uri=http://exist-db.org/xquery/xmldb#store.3.)

### Describe how to reproduce or add a test

See the query above.

### Context information

- eXist-db version: 3.3.0-DEVELOP a059dec22 (for 2.2, I tested on exist-db.org's eXide)
- Java version: 8u121
- Operating system: macOS 10.12.5
- 64 bit
- Any custom changes in e.g. conf.xml: none

### PR DESCRIPTION
**Pull request by rhubner**
This PR fixes #1456 and adds a new feature flag to disable `xs:anyURI` for `xmldb:store` and `xmldb:store-as-binary`. 

By default `xs:anyURI` is **disabled**. You can enable it with : 

```
<module uri="http://exist-db.org/xquery/xmldb" class="org.exist.xquery.functions.xmldb.XMLDBModule">
  <parameter name="anyUriDisabled" value="false"/>
</module>
```

This open source contribution to the eXist-db project was commissioned by the Office of the Historian, U.S. Department of State, https://history.state.gov/

### CODE REVIEW

**Review by adamretter — APPROVED** [review]
LGTM thanks Radek

File: exist-core/src/main/java/org/exist/xquery/functions/xmldb/XMLDBStore.java [hunk 1]
```diff
@@ -42,9 +43,11 @@
 import org.exist.util.io.TemporaryFileManager;
 import org.exist.util.serializer.SAXSerializer;
 import org.exist.xmldb.EXistResource;
+import org.exist.xquery.ErrorCodes;
 import org.exist.xquery.FunctionSignature;
 import org.exist.xquery.XPathException;
 import org.exist.xquery.XQueryContext;
+import org.exist.xquery.functions.util.UtilModule;
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/functions/xmldb/XMLDBStore.java:50]: import not used?

File: exist-core/src/main/java/org/exist/xquery/functions/xmldb/XMLDBStore.java [hunk 2]
```diff
@@ -263,7 +271,7 @@ private Resource loadFromURI(final Collection collection, final URI uri, final S
             try {
                 temp = temporaryFileManager.getTemporaryFile();
                 try(final InputStream is = uri.toURL().openStream()) {
-                    Files.copy(is, temp);
+                    Files.copy(is, temp, StandardCopyOption.REPLACE_EXISTING); //REPLACE_EXISTING because getTemporaryFile always create file on filesystem.
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/functions/xmldb/XMLDBStore.java:274]: riiiiiight you are.nice.

**Comment by dizzzz** [comment]
Nice contribution !

I find the double negation a bit confusing.... so I need to set 'anyUriDisabled' to 'false' to have something enabled?

**Review by dizzzz — APPROVED** [review]
Added one question.... #style

**Comment by rhubner** [comment]
Hello @dizzzz,

I agree that is little bit confusing. Namig is hard. Do you have any suggestion for better naming? Maybe `anyUri` or `anyUriEnabled` to remove double negative.

**Comment by dizzzz** [comment]
Sorry I missed the question; I guess "allowAnyUri" is more correct

**Comment by dizzzz** [comment]
Unfortunately the PR needs te be rebased :-(

**Pull request merged**